src/pixelator/pna/amplicon/build_amplicon.py

```diff
@@ -666,60 +708,118 @@ def _check_regions(
         :return: the created amplicon as a new SequenceRecord or None if filtered out
         """
-        # Create a slice for each region in the amplicon (if it could be found)
-        r1_regions = self._scan_forward_read(read1)
-        r2_regions = self._scan_reverse_read(read2)
+        if read1 is None and read2 is None:
+            return None
 
-        # Combine the info from forward and reverse reads
-        pid1_umi1_region_seq, pid1_umi1_region_qual = self._consensus_seq(
-            read1, read2, r1_regions.pid1_umi1, r2_regions.pid1_umi1
-        )
-        pid2_umi2_region_seq, pid2_umi2_region_qual = self._consensus_seq(
-            read1, read2, r1_regions.pid2_umi2, r2_regions.pid2_umi2
-        )
-        uei_region_seq, uei_region_qual = self._consensus_seq(
-            read1, read2, r1_regions.uei, r2_regions.uei
-        )
-        lbs1_region_qual = self._consensus_qual_lbs1(
-            read1, read2, r1_regions.lbs1, r2_regions.lbs1
-        )
-        lbs2_region_qual = self._consensus_qual_lbs2(
-            read1, read2, r1_regions.lbs2, r2_regions.lbs2
-        )
+        # Create a slice for each region in the amplicon (if it could be found)
+        r1_regions = self._scan_forward_read(read1) if read1 is not None else None
+        r2_regions = self._scan_reverse_read(read2) if read2 is not None else None
+        if read1 is None or read2 is None:
+            is_read1 = read1 is not None
+            read = read1 if is_read1 else read2
+            regions = r1_regions if is_read1 else r2_regions
+            is_reversed = not is_read1
+            pid1_umi1_region_seq, pid1_umi1_region_qual = self._get_region_sequence(
+                read,
+                regions.pid1_umi1,  # type: ignore
+                is_reversed=is_reversed,
+            )
+            pid2_umi2_region_seq, pid2_umi2_region_qual = self._get_region_sequence(
+                read,
+                regions.pid2_umi2,  # type: ignore
+                is_reversed=is_reversed,
+            )
+            uei_region_seq, uei_region_qual = self._get_region_sequence(
+                read,
+                regions.uei,  # type: ignore
+                is_reversed=is_reversed,
+            )
+            lbs1_region_qual = self._consensus_qual_lbs1(
+                read1,
+                read2,
+                region1_slice=regions.lbs1 if is_read1 else None,  # type: ignore
+                region2_slice=regions.lbs1 if not is_read1 else None,  # type: ignore
+            )
+            lbs2_region_qual = self._consensus_qual_lbs2(
+                read1,
+                read2,
+                region1_slice=regions.lbs2 if is_read1 else None,  # type: ignore
+                region2_slice=regions.lbs2 if not is_read1 else None,  # type: ignore
+            )
+            error = self._check_regions(
+                pid1_umi1_region_seq, uei_region_seq, pid2_umi2_region_seq
+            )
```

Comment:
Perhaps we can break this into a utility function? I think this function feels a bit long.